src/main/main.ts

```diff
@@ -17,7 +19,29 @@ log.transports.console.useStyles = true;
 log.transports.file.fileName = 'nova-spektr.log';
 log.transports.file.format = '{y}/{m}/{d} {h}:{i}:{s}.{ms} [{env}#{version}]-{processType} [{level}] > {text}';
 log.transports.file.level = 'info';
-log.transports.file.maxSize = 1048576 * 5; //5mb
+log.transports.file.maxSize = 1048576 * 5; //5mb;
+log.transports.file.archiveLogFn = (oldLogFile: LogFile): void => {
+  const file = oldLogFile.toString();
+  const info = path.parse(file);
+
+  try {
+    const date = new Date();
+    let newFileName = path.join(
+      info.dir,
+      info.name + '.' + date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + date.getDate() + info.ext,
+    );
+    if (fs.existsSync(newFileName)) {
+      let i: number = 1;
+      do {
+        i++;
+      } while (fs.existsSync(newFileName + '-' + i));
+      newFileName = newFileName + '-' + i;
```

Comment:
Is it Okay to add index to the end of the extension, not filename?

Reply:
It's ok. I've checked it in the article https://www.baeldung.com/java-logging-rolling-file-appenders
However it may be better to add dot instead of dash.